File: Day1/two-sum-1.py
# Two pointer approach

class Solution:
    def two_sum(self, nums, k):
        nums.sort()
        start, end = 0, len(nums) - 1
        while start < end:
            sum_val = nums[start] + nums[end]
            if sum_val == k:
                return True
            elif sum_val > k:
                end -= 1
            else:
                start += 1
        return False


def main():
    nums = [10, 15, 3, 7]
    k = 17
    solution = Solution()
    ans = solution.two_sum(nums, k)
    print(ans)


if __name__ == "__main__":
    main()

File: Day1/two-sum-1.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
	bool twoSum(vector<int> nums, int k) {
		sort(nums.begin(), nums.end());

		int start = 0, end = nums.size() - 1;
		while (start < end) {
			int sum_val = nums[start] + nums[end];
			if (sum_val == k)
				return true;
			else if (sum_val > k)
				end--;
			else
				start++;
		}
		return false;
	}
};

int main()
{
	vector<int> nums = {10, 15, 3, 7};
	int k = 17;
	Solution solution;
	bool ans = solution.twoSum(nums, k);
	cout << (ans ? "true": "false") << endl;
}
